dcs/coalition.py

```diff
@@ -156,24 +156,28 @@ def load_from_dict(self, mission, d) -> List[StatusMessage]:
 
                     # units
                     for imp_unit_idx in pgroup["units"]:
-                        imp_unit = pgroup["units"][imp_unit_idx]
-                        plane = Plane(
-                            mission.terrain,
-                            _id=imp_unit["unitId"],
-                            name=self.get_name(mission, imp_unit["name"]),
-                            _type=planes.plane_map[imp_unit["type"]],
-                            _country=_country)
-                        plane.load_from_dict(imp_unit)
-
-                        if _country.reserve_onboard_num(plane.onboard_num):
-                            msg = "{c} Plane '{p}' already using tail number: {t}".format(
-                                c=self.name.upper(), p=plane.name, t=plane.onboard_num)
-                            status.append(StatusMessage(msg, MessageType.ONBOARD_NUM_DUPLICATE, MessageSeverity.WARN))
-                            print("WARN:", msg, file=sys.stderr)
-                        status += self._park_unit_on_airport(mission, plane_group, plane)
-
-                        mission.current_unit_id = max(mission.current_unit_id, plane.id)
-                        plane_group.add_unit(plane)
+                        try:  
+                            imp_unit = pgroup["units"][imp_unit_idx]
+                            plane = Plane(
+                                mission.terrain,
+                                _id=imp_unit["unitId"],
+                                name=self.get_name(mission, imp_unit["name"]),
+                                _type=planes.plane_map[imp_unit["type"]],
+                                _country=_country)
+                            plane.load_from_dict(imp_unit)
+    
+                            if _country.reserve_onboard_num(plane.onboard_num):
+                                msg = "{c} Plane '{p}' already using tail number: {t}".format(
+                                    c=self.name.upper(), p=plane.name, t=plane.onboard_num)
+                                status.append(StatusMessage(msg, MessageType.ONBOARD_NUM_DUPLICATE, MessageSeverity.WARN))
+                                print("WARN:", msg, file=sys.stderr)
+                            status += self._park_unit_on_airport(mission, plane_group, plane)
+    
+                            mission.current_unit_id = max(mission.current_unit_id, plane.id)
+                            plane_group.add_unit(plane)
+                        except KeyError as e:
+                                msg = f"Error loading Plane, skipped loading it from miz - is it a mod?: {e}"
+                                print("ERROR", msg, file=sys.stderr)
```

Comment:
I have another concern here, if you save this .miz file, the plane group will be gone.
So I'm not sure this should go on master, and rather fixed with some new mechanic that allows to at least preserve
modded/unknown planes/helis/vehicles....

Or have some load flag, to allow best effort loading, that allows to remove unloadable data or something like that.

Reply:
The error behavior should maybe be configurable with a flag to the mission load? I went back and forth on this a few times before deleting my comment as YAGNI. I haven't decided how I'd actually want this to work in Liberation. In our case we only deserialize for campaign definitions, and if a campaign designer has put some stuff in there that we can't load, that's maybe a sign that they made a mistake and we'd be ignoring it rather than signaling that there's a bug? (yes, it'll log an error, but people don't typically read logs IME)

(in any case, @gregretkowski, if you split the rest of your changes out into a separate PR, the rest is trivially correct so I'd merge that now)

Reply:
This was my last change where I realized I wont be able to use this lib for what I wanted to do with my miz, but I got this far so I opened the PR.. I looked at how planes are defined in the lib and it looks like they are exported as very detailed data structures, afaict putting in support for them would be a large effort - and with this change above while it helps to load the mission its not going to help someone save the mission afterwards as the modded planes will be missing.

I'll remove the 'modded plane' part of this PR and just have it contain the SoundToUnit, LineStyles changes

Reply:
In that case you may want to subscribe on https://github.com/pydcs/dcs/issues/247. Not sure if it'd solve your problem or not but part of the goal is to reduce just how my pydcs needs to know about things. I haven't made any progress on that in a while, so it won't solve your problem any time soon, but it doesn't hurt to subscribe :)